src/main/java/org/opensearch/security/identity/SecurityTokenManager.java

```diff
@@ -0,0 +1,261 @@
+                    cs.getClusterName().value(),
+                    u.getName(),
+                    claims.getAudience(),
+                        claims.getExpiration(),
+                    mappedRoles,
+                    u.getRoles(),
+                    false
+                );
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }).map(BearerAuthToken::new).orElseThrow(() -> new OpenSearchSecurityException("JWT creation failed."));
+    }
+
+    /**
+     * An overridden method that creates a new ServiceAccountToken based on the provided extension ID
+     * @param extensionUniqueId The id for the extension who should have a service account token created
+     * @return A new AuthTokne in the form of a BasicAuthToken
+     * @throws OpenSearchSecurityException Thrown if the internal user account for the extension cannot be found etc.
+     */
+    @Override
+    public AuthToken issueServiceAccountToken(String extensionUniqueId) throws OpenSearchSecurityException {
+        try {
+            return new BasicAuthToken(this.userService.generateAuthToken(extensionUniqueId));
+        } catch (Exception e) {
+            throw new OpenSearchSecurityException(String.valueOf(e));
+        }
+    }
+
+    public Set<String> mapRoles(final User user, final TransportAddress caller) {
+        return this.configModel.mapSecurityRoles(user, caller);
+    }
+
+    /*
+     * The default configuration of this web key should be:
+     *   KeyType: OCTET
+     *   PublicKeyUse: SIGN
+     *   Encryption Algorithm: HS512
+     * */
+    static JsonWebKey createJwkFromSettings(Settings settings) {
+        if (!isKeyNull(settings, "signing_key")) {
+            String signingKey = settings.get("signing_key");
+
+            JsonWebKey jwk = new JsonWebKey();
+
+            jwk.setKeyType(KeyType.OCTET);
+            jwk.setAlgorithm("HS512");
+            jwk.setPublicKeyUse(PublicKeyUse.SIGN);
+            jwk.setProperty("k", signingKey);
+
+            return jwk;
+        } else {
+            Settings jwkSettings = settings.getAsSettings("jwt").getAsSettings("key");
+
+            if (jwkSettings.isEmpty()) {
+                throw new OpenSearchSecurityException(
+                        "Settings for signing key is missing. Please specify at least the option signing_key with a shared secret."
+                );
+            }
+
+            JsonWebKey jwk = new JsonWebKey();
+
+            for (String key : jwkSettings.keySet()) {
+                jwk.setProperty(key, jwkSettings.get(key));
+            }
+
+            return jwk;
+        }
+    }
+
+    public String createJwt(
+            String issuer,
+            String subject,
+            String audience,
+            Long expirySeconds,
+            Set<String> roles,
+            Set<String> backendRoles,
+            boolean roleSecurityMode
+    ) {
+        final long nowAsSecs = timeProvider.getAsLong();
```

Comment:
This may collide with the https://github.com/opensearch-project/security/pull/3421 PR where its being switched to epoch millis. We will need to stick to one convention to avoid collisions. Personally, I think seconds are a bit easier to reason with then millis but we shouldn't have a mix in different places.

Reply:
Using milliseconds is less than ideal because of how the expiration time is handled. 

If you look at the way the OnBehalfOf token requests are set up, we want people to be able to provide an expiration time that makes sense. Unfortunately, no one is going to want to provide an expiration time of 179XXXXX. So I swapped this over to be reasonable for configuration. 

I agree we will have to make sure everything is compatible.

Reply:
Hi @scrawfor99 @cwperks, if the seconds is the case, we can always *1000 in the #3421 right?